From c5191c9ec379decce4e6b2cef05fe301a22e403a Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Thu, 4 Aug 2016 17:37:46 -0400
Subject: [PATCH 0523/1722] drm/amdgpu: split dal_supported into a helper and
 wrapper

We need this to for early init to set the drm driver flags
based on the asic since not all asics support atomic right
now.

Reviewed-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu.h        |  1 +
 drivers/gpu/drm/amd/amdgpu/amdgpu_device.c | 22 +++++++++++++---------
 2 files changed, 14 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu.h b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
index 6a6c6c0..641e806 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
@@ -2169,6 +2169,7 @@ void amdgpu_io_wreg(struct amdgpu_device *adev, u32 reg, u32 v);
 
 u32 amdgpu_mm_rdoorbell(struct amdgpu_device *adev, u32 index);
 void amdgpu_mm_wdoorbell(struct amdgpu_device *adev, u32 index, u32 v);
+bool amdgpu_device_asic_has_dal_support(enum amd_asic_type asic_type);
 bool amdgpu_device_has_dal_support(struct amdgpu_device *adev);
 /*
  * Registers read & write functions.
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
index caf4717..e98b88f 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
@@ -1449,16 +1449,9 @@ static bool amdgpu_device_is_virtual(void)
 #endif
 }
 
-/**
- * amdgpu_device_has_dal_support - check if dal is supported
- *
- * @adev: amdgpu_device_pointer
- *
- * Returns true for supported, false for not supported
- */
-bool amdgpu_device_has_dal_support(struct amdgpu_device *adev)
+bool amdgpu_device_asic_has_dal_support(enum amd_asic_type asic_type)
 {
-	switch(adev->asic_type) {
+	switch (asic_type) {
 #if defined(CONFIG_DRM_AMD_DAL) && defined(CONFIG_DRM_AMD_DAL_DCE8_0)
 	case CHIP_BONAIRE:
 	case CHIP_HAWAII:
@@ -1481,6 +1474,17 @@ bool amdgpu_device_has_dal_support(struct amdgpu_device *adev)
 	}
 }
 
+/**
+ * amdgpu_device_has_dal_support - check if dal is supported
+ *
+ * @adev: amdgpu_device_pointer
+ *
+ * Returns true for supported, false for not supported
+ */
+bool amdgpu_device_has_dal_support(struct amdgpu_device *adev)
+{
+	return amdgpu_device_asic_has_dal_support(adev->asic_type);
+}
 
 /**
  * amdgpu_device_init - initialize the driver
-- 
2.7.4

